### PR TITLE
Add expiry to the top banner on adev and close the survey

### PR DESCRIPTION
**Pull request by mgechev**

### CODE REVIEW

File: adev/shared-docs/components/top-level-banner/top-level-banner.component.ts [hunk 1]
```diff
@@ -27,15 +27,21 @@ export class TopLevelBannerComponent implements OnInit {
...
-
-  // Whether the user has closed the banner.
+  // Optional expiry date
+  expiry = input<string>();
+  // Whether the user has closed the banner or the survey has expired.
   hasClosed = signal<boolean>(false);
 
   ngOnInit(): void {
+    const expiryDate = this.expiry();
+    const expired = expiryDate !== undefined && Date.now() > new Date(expiryDate).getTime();
```
Comment (MarkTechson) [inline adev/shared-docs/components/top-level-banner/top-level-banner.component.ts:37]: (nit): This doesn't handle invalid inputs, would be a nice to have?

**Review by MarkTechson — APPROVED** [review]
LGTM

(that nit I pointed out might not matter since this isn't a public API?)

**Review by MarkTechson — APPROVED** [review]
Great! Love the use of transform on the input!

LGTM 🚀

**Comment by MarkTechson** [comment]
@mgechev everything is good but there's a linting error, after that this should be merged 👍

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 7d1998fa735f67775fa740295d5d80736a4da01a.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**